lib/common/benchmarkModelGenerator.js

```diff
@@ -0,0 +1,406 @@
+            });
+            declarationI++;
+            declarations = [ ...declarations, declaration ];
+        }
+
+        return declarations;
+    }
+
+    /**
+     * Generates declarations according to a specified size budget and growth strategy.
+     * @param {Object} params An object containing model index, size budget, number of declarations, number of properties, and growth strategy.
+     * @returns {Array<Object>} An array of declaration objects according to the specified growth strategy.
+     */
+    generateDeclarationsToSize({
+        modelI,
+        declarationsSizeBudget,
+        nDeclarations,
+        nProperties,
+        growBy,
+    }) {
+        switch(growBy) {
+        case 'declarations':
+            return this.generateDeclarationsToSizeGrowByDeclarations({
+                modelI,
+                declarationsSizeBudget,
+                nProperties,
+            });
+        case 'properties':
+            return this.generateDeclarationsToSizeGrowByProperties({
+                modelI,
+                declarationsSizeBudget,
+                nDeclarations,
+            });
+        default:
+            throw new Error('growBy can be either set to "declarations" or "properties".');
+        }
+    }
+
+    /**
+     * Generates a specified number of declarations, each with a fixed number of properties.
+     * @param {Object} params An object containing model index, the number of declarations, and the number of properties per declaration.
+     * @returns {Array<Object>} An array of declaration objects.
+     */
+    generateNDeclarations({
+        modelI,
+        nDeclarations,
+        nProperties,
+    }) {
+        let declarations = [];
+
+        for (let declarationI = 0; declarationI < nDeclarations; declarationI++) {
+            declarations = [
+                ...declarations,
+                this.generateDeclarationWithNProperties({
+                    modelI,
+                    declarationI,
+                    nProperties,
+                })
+            ];
+        }
+
+        return declarations;
+    }
+
+    /**
+     * Generates a Concerto model with specified parameters, optionally generating up to a size limit.
+     * @param {Object} params An object containing parameters for the model generation, such as metamodel version, namespace, version, indices, and size constraints.
+     * @returns {Object} An object containing the generated model and metadata about the generation process.
+     * @public
+     */
+    generateConcertoModels({
+        concertoMetamodelVersion = DEFAULT_CONCERTO_METAMODEL_VERSION,
+        modelNamespace = DEFAULT_MODEL_NAMESPACE,
+        modelVersion = DEFAULT_MODEL_VERSION,
+        modelI = 0,
+        generateUpToSize,
+        growBy = 'declarations',
+        nDeclarations = 1,
+        nProperties = 1,
+    }) {
+        if (generateUpToSize > v8.getHeapStatistics().total_heap_size) {
```

Comment:
`total_available_size` would be a tighter fit

Reply:
Addressed ✅